app/modules/import-export/processes/import.php

```diff
@@ -136,33 +136,73 @@ class Import {
 	 *      (e.g: include, selected_plugins, selected_cpt, selected_override_conditions, etc.)
 	 * @throws \Exception
 	 */
-	public function __construct( string $path, array $settings = [] ) {
-		if ( is_file( $path ) ) {
-			$this->extracted_directory_path = $this->extract_zip( $path );
+	public function __construct( string $path, array $settings = [], $old_instance = null ) {
+		if ( $old_instance ) {
+			$this->set_import_object( $old_instance );
 		} else {
-			$elementor_tmp_directory = Plugin::$instance->uploads_manager->get_temp_dir();
-			$path = $elementor_tmp_directory . basename( $path );
+			if ( is_file( $path ) ) {
+				$this->extracted_directory_path = $this->extract_zip( $path );
+			} else {
+				$elementor_tmp_directory = Plugin::$instance->uploads_manager->get_temp_dir();
+				$path = $elementor_tmp_directory . basename( $path );
 
-			if ( ! is_dir( $path ) ) {
-				throw new \Exception( static::SESSION_DOES_NOT_EXITS_ERROR );
+				if ( ! is_dir( $path ) ) {
+					throw new \Exception( static::SESSION_DOES_NOT_EXITS_ERROR );
+				}
+
+				$this->extracted_directory_path = $path . '/';
 			}
 
-			$this->extracted_directory_path = $path . '/';
+			$this->session_id = basename( $this->extracted_directory_path );
+			$this->settings_referrer = ! empty( $settings['referrer'] ) ? $settings['referrer'] : 'local';
+			$this->settings_include = ! empty( $settings['include'] ) ? $settings['include'] : null;
+
+			// Using isset and not empty is important since empty array is valid option.
+			$this->settings_selected_override_conditions = $settings['overrideConditions'] ?? null;
+			$this->settings_selected_custom_post_types = $settings['selectedCustomPostTypes'] ?? null;
+			$this->settings_selected_plugins = $settings['plugins'] ?? null;
+
+			$this->manifest = $this->read_manifest_json();
+			$this->site_settings = $this->read_site_settings_json();
+
+			$this->set_default_settings();
 		}
+	}
+
+	private function set_import_object( $old_instance ) {
```

Comment:
please change the name of the $old_instance to $instance -
thi function don't need to know that this is new or old instance.

Reply:
Done